### DISCUSSION TITLE
How do Page Image Resource work with the images module

### DISCUSSION THREAD

**Discussion by krischik** [hugomods/.github#1] [Q&A] [answered]
I have set up the images modules and I got „Site Image Resources“ to work. However I would prefer to use „Page Image Resource“. According to [the documentation](https://hugomods.com/en/docs/images/) that should be possible.

However, no matter where I put the images all I get is

```sh
WARN  image not found: Opcode.png
WARN  image not found: Vertors.png
WARN  image not found: Main.png
WARN  image not found: Screenshot_at_Jul_16_18-35-24.png?height=320px
WARN  image not found: Screenshot_at_Jul_16_18-36-18.png?height=329px
WARN  image not found: Screenshot_at_Jul_16_18-37-15.png?width=640px
```

I have tried the `static`, `assert` and `content` directory. None worked. I did changed one file to all lower caps as hugo has problems with upper case characters. That didn't make a difference either.

I have searched github for the `github.com/hugomods/images` for samples but while there are lost of users of the module no one seem to have lots of images files organised in pages.

Full source in `https://git.code.sf.net/p/tutorial-6502/code` branch `feature/scale_images`

**Comment by razonyang** [comment, marked as answer]
Hi @krischik, your code service seems private, it returns 404.

To use the [page's images resources](https://gohugo.io/content-management/page-resources/), you'll need to organize your pages as [page bundles](https://gohugo.io/content-management/page-bundles/). For example.

```sh
content/
  blog/
    hello/
      index.md
      sample.png
```

There is a `/blog/hello` page which contains one image resource, then you can access the image in the content (`index.md`).

```markdown
![Sample](sample.png)
```

The real example is the images module docs, it uses the image page resource (`featured.jpeg`) as example.

- Source: https://github.com/hugomods/site/tree/main/main/content/docs/images
- Site: https://hugomods.com/en/docs/images

**Reply by krischik** [reply]
Not private but probably the wrong link. Try

```sh 
git clone git://git.code.sf.net/p/tutorial-6502/code tutorial-6502-code
```
Or just browse it here:

https://sourceforge.net/p/tutorial-6502/code/ci/master/tree/

**Comment by krischik** [comment]
> To use the [page's images resources](https://gohugo.io/content-management/page-resources/), you'll need to organize your pages as [page bundles](https://gohugo.io/content-management/page-bundles/). For example.

That was my first attempt as well. But thanks for the [page bundles](https://gohugo.io/content-management/page-bundles/) link. I was struggling with the `index.md` vs `_index.md`. I had to figure it out by try and error so having a proper documentation might help.

**Reply by razonyang** [reply]
You're right, I just updated the docs with more details and examples.

**Comment by krischik** [comment]
Thanks. The following successfully converted one page:

```sh
cd content/atari/hello_world
mkdir car_using_assembler   
git mv car_using_assembler.md car_using_assembler/index.md
git mv ../../../static/atari/hello_world/car_using_assembler/* car_using_assembler 
```

The rest ist just diligence.
